NitroxClient/GameLogic/SimulationOwnership.cs

```diff
@@ -67,14 +111,76 @@ public void ReceivedSimulationLockResponse(NitroxId id, bool lockAquired, Simula
             }
         }
 
+        public void RemovePlayerSimulations(ushort playerId)
+        {
+            // When a player disconnects, we need to remove his locks
+            // We also remove locks of unknown player, as they could come from the disconnected player
+            HashSet<NitroxId> toRemove = new HashSet<NitroxId>();
+            foreach (KeyValuePair<NitroxId, PlayerLock> playerLock in simulatedIdsByLockType)
+            {
+                if (!playerLock.Value.PlayerId.HasValue ||playerLock.Value.PlayerId == playerId)
+                {
+                    toRemove.Add(playerLock.Key);
+                }
+            }
+            foreach (NitroxId id in toRemove)
+            {
+                simulatedIdsByLockType.Remove(id);
+            }
+        }
+
         public void SimulateEntity(NitroxId id, SimulationLockType lockType)
         {
-            simulatedIdsByLockType[id] = lockType;
+            simulatedIdsByLockType[id] = new PlayerLock(muliplayerSession.Reservation.PlayerId, lockType);
         }
 
         public void StopSimulatingEntity(NitroxId id)
         {
             simulatedIdsByLockType.Remove(id);
         }
+        
+        public void AddSimulationLock(NitroxId id, ushort playerId, SimulationLockType lockType)
+        {
+            simulatedIdsByLockType[id] = new PlayerLock(playerId, lockType);
+        }
+
+        public void AddSimulationLockForUnknown(NitroxId id, SimulationLockType lockType)
+        {
+            simulatedIdsByLockType[id] = new PlayerLock(null, lockType);
+        }
+
+        public void AddSimulationOverride(NitroxId id)
+        {
+            if (simulatedIdsByLockType.ContainsKey(id) && simulatedIdsByLockType[id].PlayerId == muliplayerSession.Reservation.PlayerId)
```

Comment:
Use `simulatedIdsByLockType.TryGetValue(id, out ...)` to reduce amount of lookups. Then just keep the ` && simulatedIdsByLockType[id].PlayerId == muliplayerSession.Reservation.PlayerId` as is.